crypto/ed25519/ed25519.go

```diff
@@ -16,50 +16,33 @@ const PrivateKeySize = ed25519.PrivateKeySize
 const SignatureSize = ed25519.SignatureSize
 const SeedSize = ed25519.SeedSize
 
-func GenKeyPair() ([]byte, *big.Int, *big.Int, error) {
-	pubKey, privKey, err := ed25519.GenerateKey(rand.Reader)
-	if err != nil {
-		return nil, nil, nil, errors.New("NewEd25519: Generate key pair error")
-	}
-
-	X := new(big.Int).SetBytes(pubKey)
-	Y := big.NewInt(0)
-
-	return privKey, X, Y, nil
+func GenKeyPair() ([]byte, []byte, error) {
+	return ed25519.GenerateKey(rand.Reader)
 }
 
-func Sign(priKey []byte, data []byte) (*big.Int, *big.Int, error) {
-	sig := ed25519.Sign(priKey, data)
-	return new(big.Int).SetBytes(sig[:ed25519.SignatureSize/2]), new(big.Int).SetBytes(sig[ed25519.SignatureSize/2:]), nil
+func Sign(privateKey, data []byte) ([]byte, error) {
+	if len(privateKey) != PrivateKeySize {
```

Comment:
I think `ed25519.Sign` will check privatekey length itself?

Reply:
`ed25519.Sign` will panic if length is incorrect, and we don't want panic here.